dcos_installer/upgrade.py

```diff
@@ -38,6 +38,15 @@
 
 source /opt/mesosphere/environment.export
 
+# bash should trap the ERR signal and run err_report, which prints the line of error
+# to STDERR.
+
+err_report() {
```

Comment:
This trick is from here: http://stackoverflow.com/a/42219754/18852

Short snippets fall under fair use clause. (I can't think of a better way ignoring this).